aten/src/ATen/core/Tensor.h

```diff
@@ -1,3 +1,33 @@
 #pragma once
 
 #include <ATen/core/TensorBody.h>
+
+namespace at {
+class TORCH_API OptionalTensorRef {
+ public:
+  OptionalTensorRef() {}
+  OptionalTensorRef(const Tensor& src)
+      : ref_(c10::intrusive_ptr<TensorImpl>(
+            src.unsafeGetTensorImpl(),
+            c10::raw::DontIncreaseRefcount{})) {}
```

Comment:
For extra safety, assert src is NOT undefined here.

Reply:
I don't see the assert here?

Reply:
Oops, got deleted in my last revision for some reason. I'll add it back.